h/services/search_index/_queue.py

```diff
@@ -160,43 +162,39 @@ def sync(self, limit):
             if job.kwargs.get("force", False):
                 annotation_ids_to_sync.add(annotation_id)
                 job_complete.append(job)
-                counts[Queue.Result.SYNCED_FORCED] += 1
-                counts[Queue.Result.COMPLETED_FORCED] += 1
+                counts[Queue.Result.SYNCED_FORCED.format(tag=job.tag)] += 1
+                counts[Queue.Result.SYNCED_TAG_TOTAL.format(tag=job.tag)] += 1
+                counts[Queue.Result.SYNCED_TOTAL] += 1
+                counts[Queue.Result.COMPLETED_FORCED.format(tag=job.tag)] += 1
+                counts[Queue.Result.COMPLETED_TAG_TOTAL.format(tag=job.tag)] += 1
+                counts[Queue.Result.COMPLETED_TOTAL] += 1
             elif not annotation_from_db:
                 job_complete.append(job)
-                counts[Queue.Result.COMPLETED_DELETED] += 1
+                counts[Queue.Result.COMPLETED_DELETED.format(tag=job.tag)] += 1
+                counts[Queue.Result.COMPLETED_TAG_TOTAL.format(tag=job.tag)] += 1
+                counts[Queue.Result.COMPLETED_TOTAL] += 1
             elif not annotation_from_es:
                 annotation_ids_to_sync.add(annotation_id)
-                counts[Queue.Result.SYNCED_MISSING] += 1
+                counts[Queue.Result.SYNCED_MISSING.format(tag=job.tag)] += 1
+                counts[Queue.Result.SYNCED_TAG_TOTAL.format(tag=job.tag)] += 1
+                counts[Queue.Result.SYNCED_TOTAL] += 1
             elif not self._equal(annotation_from_es, annotation_from_db):
                 annotation_ids_to_sync.add(annotation_id)
-                counts[Queue.Result.SYNCED_DIFFERENT] += 1
+                counts[Queue.Result.SYNCED_DIFFERENT.format(tag=job.tag)] += 1
+                counts[Queue.Result.SYNCED_TAG_TOTAL.format(tag=job.tag)] += 1
+                counts[Queue.Result.SYNCED_TOTAL] += 1
             else:
                 job_complete.append(job)
-                counts[Queue.Result.COMPLETED_UP_TO_DATE] += 1
+                counts[Queue.Result.COMPLETED_UP_TO_DATE.format(tag=job.tag)] += 1
+                counts[Queue.Result.COMPLETED_TAG_TOTAL.format(tag=job.tag)] += 1
+                counts[Queue.Result.COMPLETED_TOTAL] += 1
```

Comment:
It's funny how each case in the `if` / `else` has 1-2 lines of actual code (mostly 1) and then 3-6 lines of recording metrics. I like it though because it's dumb as a post: each case is just flat code that takes the necessary actions and updates the necessary counts. It's very easy to read it and verify that it's right. I prefer it over separately computing the totals counts like it did before (below)